### PR TITLE
Fix AddStaticFiles method

### PR DESCRIPTION
**Pull request by Umang01-hash**
## Pull Request Template

**Description:**
- Resolve issue #1146 
-   Move the setup the default routes of `/.well-known` to `New` method of app, instead of `Run` so as to ensure proper handlers are called for the specific routes instead of **staticHandler**.

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

File: pkg/gofr/gofr.go [hunk 1]
```diff
@@ -88,6 +88,17 @@ func New() *App {
 	app.httpServer.certFile = app.Config.GetOrDefault("CERT_FILE", "")
 	app.httpServer.keyFile = app.Config.GetOrDefault("KEY_FILE", "")
 
+	// Add Default routes
+	app.add(http.MethodGet, "/.well-known/health", healthHandler)
+	app.add(http.MethodGet, "/.well-known/alive", liveHandler)
```
Comment (ccoVeille) [inline pkg/gofr/gofr.go:93]: Just a question, not about your changes in this PR

Are these well known URI somehow standard?

The only documentation I found was the gofr documentation

https://gofr.dev/docs/advanced-guide/monitoring-service-health

The documentation doesn't talk about it

I don't find them also on Wikipedia

https://en.wikipedia.org/wiki/Well-known_URI

Doesn't it mean gofr created its own "standard", but that is the only one to use?

I'm unsure I would have used .well-known URI for these

Unless there is an rfc or other project using this

File: pkg/gofr/http/router.go [hunk 2]
```diff
@@ -56,7 +59,12 @@ func (rou *Router) AddStaticFiles(endpoint, dirName string) {
 	cfg := staticFileConfig{directoryName: dirName}
 
 	fileServer := http.FileServer(http.Dir(cfg.directoryName))
-	rou.Router.NewRoute().PathPrefix(endpoint + "/").Handler(http.StripPrefix(endpoint, cfg.staticHandler(fileServer)))
+
+	if endpoint == "/" {
+		rou.Router.NewRoute().PathPrefix("/").Handler(cfg.staticHandler(fileServer))
+	} else {
+		rou.Router.NewRoute().PathPrefix(endpoint + "/").Handler(http.StripPrefix(endpoint, cfg.staticHandler(fileServer)))
+	}
```
Comment (ccoVeille) [inline pkg/gofr/http/router.go:67]: What about this

```suggestion
	if endpoint == "/" {
		rou.Router.NewRoute().PathPrefix("/").Handler(cfg.staticHandler(fileServer))
		return
	}
	rou.Router.NewRoute().PathPrefix(endpoint + "/").Handler(http.StripPrefix(endpoint, cfg.staticHandler(fileServer)))
```

Comment (ccoVeille) [inline pkg/gofr/http/router.go:67]: I'm unsure it's the right place (lines of code or even the right PR) to comment this, but I'm commenting here, when I think about it.

If my inputs lead to a conclusion, please consider opening an issue.

So here is my point:

Are we sure the URL path is cleaned?

I found many framework having issues when a path in URL was provided with `//`

Like `//favicon.ico` or `/.well-known//health` it's often caused by a URL of the server containing a trailing slash in a environment variable.

This often leads to 404, while an Apache2 or NGinx server would have served the content

My remark is valid for static files but also for API endpoints

File: pkg/gofr/gofr.go [hunk 3]
```diff
@@ -88,6 +88,17 @@ func New() *App {
...
+	app.add(http.MethodGet, "/favicon.ico", faviconHandler)
+
+	if _, err = os.Stat("./static/openapi.json"); err == nil {
+		app.add(http.MethodGet, "/.well-known/openapi.json", OpenAPIHandler)
+		app.add(http.MethodGet, "/.well-known/swagger", SwaggerUIHandler)
+		app.add(http.MethodGet, "/.well-known/{name}", SwaggerUIHandler)
```
Comment (ccoVeille) [inline pkg/gofr/gofr.go:99]: This line requires a comment

Is it a kind of catchall to send anything that is not openapi.json and swagger to the SwaggerUIHandler

File: pkg/gofr/gofr.go [hunk 4]
```diff
@@ -88,6 +88,17 @@ func New() *App {
 	app.httpServer.certFile = app.Config.GetOrDefault("CERT_FILE", "")
 	app.httpServer.keyFile = app.Config.GetOrDefault("KEY_FILE", "")
 
+	// Add Default routes
+	app.add(http.MethodGet, "/.well-known/health", healthHandler)
+	app.add(http.MethodGet, "/.well-known/alive", liveHandler)
+	app.add(http.MethodGet, "/favicon.ico", faviconHandler)
+
+	if _, err = os.Stat("./static/openapi.json"); err == nil {
+		app.add(http.MethodGet, "/.well-known/openapi.json", OpenAPIHandler)
```
Comment (ccoVeille) [inline pkg/gofr/gofr.go:97]: Why not using the default swagger constant you defined? For these 2 lines.

Comment (Umang01-hash) [reply pkg/gofr/gofr.go]: Sure used the constant here

File: pkg/gofr/gofr.go (hunk 3 above)
Comment (Umang01-hash) [reply pkg/gofr/gofr.go:104]: Sure added the comments

File: pkg/gofr/http/router.go (hunk 2 above)
Comment (Umang01-hash) [reply pkg/gofr/http/router.go]: Sure changed the code as per your suggestion

File: pkg/gofr/gofr.go (hunk 1 above)
Comment (Umang01-hash) [reply pkg/gofr/gofr.go:93]: Hi @ccoVeille 

Thank you for your insightful question regarding the `.well-known` URIs.

You are correct that these URIs are not part of an official RFC or standard for the specific use cases we've implemented in GoFr. We developed these endpoints to streamline access to important API information such as health status, swagger documentation, and more. Our goal was to provide a consistent and convenient way for developers to obtain vital information about their APIs without the need for additional authentication.

While the .well-known URIs in GoFr may not align with a globally recognized standard, we believe they offer a practical solution for common development needs. We appreciate your feedback and will continue to consider how we can improve our approach to be more in line with industry standards and best practices.

Please let us know if you have any further questions or suggestions. We're here to help!

Comment (ccoVeille) [reply pkg/gofr/gofr.go:93]: I think this information is solid. I would suggest to add them in the code documentation but also in the official documentation

File: pkg/gofr/http/router.go (hunk 2 above)
Comment (Umang01-hash) [reply pkg/gofr/http/router.go]: Sure will add that issue

File: pkg/gofr/gofr.go (hunk 1 above)
Comment (Umang01-hash) [reply pkg/gofr/gofr.go:93]: Sure @ccoVeille will add this to the documentations also

Comment (ccoVeille) [reply pkg/gofr/gofr.go:93]: Side remark, I would have used a gofr prefix in the URI then
Something like /.well-known/gofr

But I assume it's either too late and opinionated

**Review by ccoVeille — APPROVED** [review]

**Review by vipul-rawat — APPROVED** [review]

**Pull request merged**